wagtail/contrib/modeladmin/templatetags/modeladmin_tags.py

```diff
@@ -71,13 +71,15 @@ def items_for_result(view, result, request):
         row_attrs = modeladmin.get_extra_attrs_for_field_col(result, field_name)
         row_attrs["class"] = " ".join(row_classes)
         row_attrs_flat = flatatt(row_attrs)
+        primary_button = None
         if field_name == modeladmin.get_list_display_add_buttons(request):
-            edit_url = view.url_helper.get_action_url("edit", result.pk)
+            primary_button = view.button_helper.get_primary_button_for_obj(result)
```

Comment:
Now that we're decoupled the method from `get_buttons_for_obj()` quite a bit more, how do you feel about the originally proposed `get_primary_action_link()` name for this method? If not, I think we should at least consider losing the `_for_obj` from the method name, since that's probably apparent enough when `obj` is the only argument.

Reply:
I would keep the name, as it also returns those button-dicts, which makes it clearer what data to expect.
But we can strip `_for_obj`, good point!